vita3k/https/src/https.cpp

```diff
@@ -236,285 +65,77 @@ std::string get_web_regex_result(const std::string &url, const std::regex &regex
-        if (std::all_of(file_size_str.begin(), file_size_str.end(), ::isdigit))
-            file_size = std::stoll(file_size_str);
-    } else
-        LOG_ERROR("No success found regex: {}", header);
-
-    return file_size;
-}
-
-static std::string convert_md5_bytes_to_str(unsigned char *md5_bytes) {
-    std::string content_md5(MD5_DIGEST_LENGTH * 2, '\0');
-    for (int i = 0; i < MD5_DIGEST_LENGTH; i++) {
-        sprintf(&content_md5[i * 2], "%02X", md5_bytes[i]);
-    }
-
-    return content_md5;
-}
-
-static std::string calculate_md5_file(const std::string &file_path) {
-    std::ifstream file(file_path, std::ios::binary);
-    if (!file) {
-        LOG_ERROR("Failed to open file: {}", file_path);
-        return {};
-    }
-
-    MD5_CTX md5_context;
-    MD5_Init(&md5_context);
-
-    char buffer[1024];
-    while (!file.eof()) {
-        file.read(buffer, sizeof(buffer));
-        MD5_Update(&md5_context, buffer, file.gcount());
-    }
+uint64_t g_start_time;
+std::function<ProgressState(float, uint64_t)> g_progress_callback;
 
-    unsigned char md5_digest[MD5_DIGEST_LENGTH];
-    MD5_Final(md5_digest, &md5_context);
+uint64_t get_current_time_ms() {
+    return std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::high_resolution_clock::now().time_since_epoch()).count();
+};
 
-    return convert_md5_bytes_to_str(md5_digest);
+static size_t write_data(void *ptr, size_t size, size_t nmemb, void *stream) {
+    size_t written = fwrite(ptr, size, nmemb, (FILE *)stream);
+    return written;
 }
 
-static std::string get_content_md5(const std::string &header) {
-    std::smatch match;
-    std::string content_md5_base64;
-    // Get the redirection URL from the response header (Location)
-    if (std::regex_search(header, match, std::regex("Content-MD5: ([-A-Za-z0-9+/=]+)"))) {
-        content_md5_base64 = match[1];
-    } else {
-        LOG_ERROR("No success found Content-MD5:\n{}", header);
-        return {};
-    }
-
-    const std::string base64_chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
-
-    unsigned char md5_bytes[MD5_DIGEST_LENGTH];
-    int i = 0;
-    uint32_t accum = 0;
-    int accum_bits = 0;
-
-    for (const auto c : content_md5_base64) {
-        if (std::isspace(c) || (c == '=')) {
-            continue;
-        }
-
-        size_t char_value = base64_chars.find(c);
-        if (char_value == std::string::npos) {
-            LOG_ERROR("Invalid character in base64 string: {}", c);
-            return {};
-        }
-
-        accum = static_cast<uint32_t>(accum << 6) | char_value;
-        accum_bits += 6;
-
-        if (accum_bits >= 8) {
-            accum_bits -= 8;
-            md5_bytes[i++] = static_cast<unsigned char>((accum >> accum_bits) & 0xFF);
-        }
-    }
+typedef int (*CurlDownloadCallback)(void *bar, double t, double d);
 
-    return convert_md5_bytes_to_str(md5_bytes);
-}
+CURL *curl_download;
```

Comment:
this is wrong, here you set general value for only one download in same time
my current code of https allow multi download in same time